reducer/last.go

```diff
@@ -18,8 +18,16 @@ func (lp *LastProto) Instantiate() Interface {
 	return &Last{Resolver: lp.resolver}
 }
 
-func NewLastProto(target string, resolver expr.FieldExprResolver) *LastProto {
-	return &LastProto{target, resolver}
+func (lp *LastProto) TargetResolver() expr.FieldExprResolver {
+	return lp.tresolver
+}
+
+func NewLastProto(target string, tresolver, resolver expr.FieldExprResolver) *LastProto {
+	return &LastProto{
+		target:    target,
+		tresolver: tresolver,
+		resolver:  resolver,
+	}
```

Comment:
We usually put constructors immediately after the type definition, not after its methods.